conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -96,7 +96,7 @@ def template(self):
             set_property(TARGET {{root_target_name}}
                          PROPERTY INTERFACE_LINK_LIBRARIES
                          $<$<CONFIG:{{configuration}}>:{{ '${'+pkg_name+'_OBJECTS'+config_suffix+'}' }}>
-                         ${{'{'}}{{pkg_name}}_LIBRARIES_TARGETS}
+                         $<$<CONFIG:{{configuration}}>:${{'{'}}{{pkg_name}}_LIBRARIES_TARGETS}>
```

Comment:
So now we have to "protect" these library targets with the configuration

Reply:
Nice, this is exactly what I meant! Doing this by locally modifying generated cmake files is exactly how I worked around the issue ^^